crates/codegen/schema/src/types/precedence_parser.rs

```diff
@@ -0,0 +1,51 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use super::parser::ParserDefinition;
+
+#[allow(dead_code)]
+pub type PrecedenceParserRef = std::rc::Rc<PrecedenceParser>;
+
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq, Eq, Hash)]
+#[serde(deny_unknown_fields)]
+pub struct PrecedenceParser {
+    #[serde(default)]
+    pub name: Option<String>,
+    #[serde(flatten)]
+    pub definition: PrecedenceParserDefinition,
```

Comment:
I wonder why `PrecedenceParserDefinition` is defined as an external `struct` and then flattened into the only parent, instead of just absorbing into the parent. Do we need that indirection here?